features/steps/steps.py

```diff
@@ -413,3 +485,64 @@ def step_impl(context, entity, other_entities):
                 errors.append(instance_structure_error(inst, [i for i in nested_entities if i.is_a() in differences], 'nested by'))
     
     handle_errors(context, errors)
+
+def get_duplicate_values(values):
+    seen = set()
+    duplicates = [x for x in values if x in seen or seen.add(x)]
+    return duplicates
+
+def ifcopenshell_instance_type_to_string(v):
+    """
+        Converts ifcopenshell instance type to strings, if applicable
+    """
+    return do_try(lambda: v[0].is_a(), v)
+
+def empty_tuple_to_string(v):
+    """
+        Converts empty tuples type to strings, if applicable
+        To be used for meaningful error messages
+    """
+    return 'None' if isinstance(v, tuple) and not v else v
+
+def map_many(v, fn, *args):
+    """
+        Maps multiple functions to a list
+    """
+    return map_many(map(fn, v), *args) if args else map(fn, v)
+
+@then("The values must be {identical_or_unique}")
+def step_impl(context, identical_or_unique):
+    errors = []
+
+    if getattr(context, 'applicable', True):
+
+        stack_tree= list(filter(None, list(map(lambda layer: layer.get('instances'), context._stack))))
+
+        for i, values in enumerate([context.instances] if getattr(context, 'within_model', False) else context.instances):
+            if not values:
+                continue
+                        
+            values_str = list(map_many(values, empty_tuple_to_string, ifcopenshell_instance_type_to_string))
+            attribute = getattr(context, 'attribute', 'None')
```

Comment:
I'd prefer `None` and not `"None"` so that we can meaningfully distinguish this case in the formatting.

Reply:
https://github.com/buildingSMART/ifc-gherkin-rules/pull/35/commits/bedf3c266e7248420159dec3e15c4c2daf75c754